src/app/_services/depublication.service.ts

```diff
@@ -0,0 +1,129 @@
+  */
+  handleUploadEvents(event: HttpEvent<Object>): boolean {
+    if (event.type === HttpEventType.Response) {
+      return true;
+    } else {
+      if (event.type === HttpEventType.UploadProgress) {
+        console.log(`File upload: ${event.loaded} / ${event.total}`);
+      }
+      return false;
+    }
+  }
+
+  /** setPublicationFile
+  /*  post publication information
+  /*  @param {string} datasetId - the dataset id
+  /*  @param {File} file - file of record urls
+  */
+  setPublicationFile(datasetId: string, file: File): Observable<boolean> {
+    const url = `${apiSettings.apiHostCore}/depublished_records/${datasetId}`;
+    const formData = new FormData();
+
+    formData.append('depublicationFile', file);
+
+    return this.http
+      .post(url, formData, {
+        headers: {
+          'Content-Type': 'multipart/form-data'
+        },
+        observe: 'events',
+        params: {
+          clientFilename: file.name,
+          mimeType: file.type
+        },
+        reportProgress: true
+      })
+      .pipe(map(this.handleUploadEvents))
+      .pipe(this.errors.handleRetry());
+  }
+
+  /** setPublicationInfo
+  /*  post publication information
+  /*  @param {string} datasetId - the dataset id
+  /*  @param {string} toDepublish - depublication record urls
+  */
+  setPublicationInfo(datasetId: string, toDepublish: string): Observable<boolean> {
+    const url = `${apiSettings.apiHostCore}/depublished_records/${datasetId}`;
+    return this.http
+      .post<boolean>(
+        url,
+        { toDepublish },
+        {
+          headers: {
+            'Content-Type': 'text/plain'
+          }
+        }
+      )
+      .pipe(this.errors.handleRetry());
+  }
+
+  /** parseFilterParameter
+  /*  return a url parameter representation of a filter string or an empty string
+  /*  @param {string} p - optional
+  */
+  parseFilterParameter(p?: string): string {
+    return p ? `&searchQuery=${p}` : '';
+  }
+
+  /** parseSortParameter
+  /*  return a url parameter representation of a SortParameter or an empty string
+  /*  @param {SortParameter} p - optional
+  */
+  parseSortParameter(p?: SortParameter): string {
+    return p ? `&sortField=${p.field}&sortDirection=${p.direction.toLowerCase()}` : '';
+  }
+
+  /** getPublicationInfo
+  /*  retrieve publication information
+  /*  @param {string} datasetId - the dataset id
+  /*  @param {number} page - the pagination target
+  /*  @param {SortParameter} sort - optional - sort parameter
+  /*  @param {string} filter - optional - sort parameter
+  */
+  getPublicationInfo(
+    datasetId: string,
+    page: number,
+    sort?: SortParameter,
+    filter?: string
+  ): Observable<Results<RecordPublicationInfo>> {
+    const sortParam = this.parseSortParameter(sort);
+    const filterParam = this.parseFilterParameter(filter);
+    const url = `${apiSettings.apiHostCore}/depublished_records/${datasetId}?page=${page}${sortParam}${filterParam}`;
```

Comment:
Would this not run the risk of invalid urls? Should the filter string not be escaped for URL parameters?
Perhaps there is some sort of URL builder we can use so that we don't have to actually manually create a query string?

Reply:
used `encodeURIComponent()`